### PR TITLE
Article - Enums in elixir

### PR DESCRIPTION
**Pull request by taher435**
Article about using enums (like rails) in Elixir

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ owner:
   codepen: #username
   hiring:
     status: true
-    link: https://honestbee.sg/en/careers
+    link: https://boards.greenhouse.io/honestbee
```
Comment (mingsheng88) [inline _config.yml]: this should be an individual pr 😁

File: _layouts/post.html [hunk 2]
```diff
@@ -65,10 +65,10 @@ <h1 title="{{ page.title }}">{{ page.title }}</h1>
   <article class="post-content">
     {{ content }}
   </article>
-  
+
   <p>
     <strong>
-      Like what you see? Come join us and be part of an amazing team. <a href="https://honestbee.sg/en/careers" target="_blank">We are hiring</a>
+      Like what you see? Come join us and be part of an amazing team. <a href="{{ site.owner.hiring.link }}" target="_blank">We are hiring</a>
```
Comment (mingsheng88) [inline _layouts/post.html]: 😁

File: _posts/2017-04-28-enums-in-elixir-ecto.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
+---
+layout: post
+title:  "Using Enums (like Rails) in Elixir"
+excerpt: "Ecto in Elixir does not support enums which is heavily used in Rails. But using macro, we can create custom enums"
+categories: [elixir]
+tags: [elixir,ecto,phoenixframework]
+author: Taher Dhilawala
+---
+
+We recently started working with Elixir (we are also [hiring](https://boards.greenhouse.io/honestbee/jobs/669258#.WQI1OomGNE4) for that btw :)). Most of the projects in honestbee are done in Rails and so majority of us here are Rubyist. When I first started working with Elixir, almost all of initial Google searches where like "Rails X equivalent in Elixir". To my surprise, there are a lot of people like me :).
```
Comment (mingsheng88) [inline _posts/2017-04-28-enums-in-elixir-ecto.md]: is there a better way to manage links? i mean 3 months down, we wouldn't be updating this value in the article. will this become a broken link?

File: _config.yml (hunk 1 above)
Comment (lenguyenthedat) [inline _config.yml]: https://honestbee.sg/en/careers looks nicer though

Comment (taher435) [inline _config.yml]: @mingsheng88 eh :trollface: ?

File: _posts/2017-04-28-enums-in-elixir-ecto.md (hunk 3 above)
Comment (lenguyenthedat) [inline _posts/2017-04-28-enums-in-elixir-ecto.md]: The article started with `We` but for the rest you used `I`, I think it's a bit inconsistent.

File: _config.yml (hunk 1 above)
Comment (taher435) [inline _config.yml]: @lenguyenthedat hmm, when I changed it the link did not show our recent roles so I thought it's no longer being maintained/updated. But now I see it has all the roles listed. Will change it back.

File: _posts/2017-04-28-enums-in-elixir-ecto.md (hunk 3 above)
Comment (taher435) [inline _posts/2017-04-28-enums-in-elixir-ecto.md]: We can use an URL shortener and manage the link redirection from there. Once this job expires, that shortened link can redirect to our jobs home page.

Comment (lenguyenthedat) [inline _posts/2017-04-28-enums-in-elixir-ecto.md]: I think this sentence is a bit hard to follow, removing the recruitment link here can make it better (should put it at the end if you want).

Would it be better to start with: "At honestbee, most of the projects in honestbee are done in Rails", and put a small explanation on why you are exploring Elixir?

File: _posts/2017-04-28-enums-in-elixir-ecto.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    _ ->
+      quote do
+        {:error, "Only maps allowed as values of enum"}
+      end
+  end
+```
+
+If the case statement above looks alien, then read up a bit about [case](http://elixir-lang.org/getting-started/case-cond-and-if.html#case) and [pattern matching](http://elixir-lang.org/getting-started/pattern-matching.html) in elixir.
+
+Remember that everything in elixir is represented as a three element tuple? Our do block is passed to our macro function as a tuple with the third element as our map in a keyword list (the check in the when clause). So if the format is correct, we pattern match and fetch the value map which is used as a return value of the function. If we pass something else other than a map, the pattern matching fails and the enum value is set with an error tuple. So when you try to fetch the enum value, you will get an error.
```
Comment (lenguyenthedat) [inline _posts/2017-04-28-enums-in-elixir-ecto.md]: s/elixir/Elixir/g

**Comment by mingsheng88** [comment]
Left my initial comments, will read and leave comments on the actual article later

File: _config.yml (hunk 1 above)
Comment (lenguyenthedat) [inline _config.yml]: I think best to check with the recruiting team + @albertojg on that 🤔

**Comment by lenguyenthedat** [comment]
LGTM (I'm not the domain expert though) 😬

**Comment by taher435** [comment]
@lenguyenthedat can you approve the PR?
